src/Store.h

```diff
@@ -150,11 +150,11 @@ class StoreEntry : public hash_link, public Packable
     void dump(int debug_lvl) const;
     void hashDelete();
     void hashInsert(const cache_key *);
-    void registerAbort(STABH * cb, void *);
+    void registerAbort(const AsyncCall::Pointer &);
     void reset();
     void setMemStatus(mem_status_t);
     bool timestampsSet();
-    void unregisterAbort();
+    void unregisterAbort(const char *reason);
```

Comment:
```suggestion
    /// Do not notify anybody about a 3rd-party initiated StoreEntry abort.
    /// Calling this method does not cancel the already queued notification.
    /// TODO: Refactor to represent the end "ownership" by our writer.
    void unregisterAbort(const char *reason);
```

Justification for that TODO: The existing registerAbortCallback() callers always unlock the entry -- today, this method signals the end of writer's ownership of the StoreEntry. Calling unregisterAbort() does not prevent an already queued callback from being fired so any future hypothetical unregisterAbort() caller that does _not_ end its StoreEntry ownership will not be able to rely on that cancellation anyway -- the method is called "unregister", but it is a rather poor mechanism for ending notifications. Together with the surrounding code in callers, this method is (implicitly) used for another purpose -- writing ownership termination.

If you agree, please add that justification to the branch commit message that adds that TODO. I realize that it will not make it into the official code, but that is the best record I can think of without creating a dedicated PR just for this TODO.